airflow/providers/hashicorp/_internal_client/vault_client.py

```diff
@@ -103,7 +108,11 @@ def __init__(
         password: str | None = None,
         key_id: str | None = None,
         secret_id: str | None = None,
+        session_token: str | None = None,
+        header_value: str | None = None,
         role_id: str | None = None,
+        use_token: bool = True,
+        region: str = "us-east-1",
```

Comment:
I'm know that hvac client use `us-east-1` as default region 🤦 🤦‍♂️ 🤦‍♀️ 

https://github.com/hvac/hvac/blob/6e47552b12573bb150a042922031bf95e4231759/hvac/aws_utils.py#L8

In general by default it is not a good idea redirect to US region, especially if customer from other regions.
I would suggest to set it by default to None and fallback from environment variables, which commonly uses in different AWS SDKs (`AWS_REGION` in all SDKs, and `AWS_DEFAULT_REGION` in addition in botocore/boto and AWS CLI) and provide it conditionally

```python
region = os.environ.get("AWS_REGION") or os.environ.get("AWS_DEFAULT_REGION") or "us-east-1"
```

Reply:
Agree, I'll change this one.

Reply:
Since I have reverted this particular change, this feedback does not apply now.